### ISSUE TITLE
compile error

### ISSUE THREAD

**Issue by NeilTruter** [eclipse-cyclonedds/cyclonedds-cxx#370]
When compiling cyclonedds-cxx on ubuntu as per cyclonedds documentation I get the following error:

/home/<user>/Projects/Tools/cyclonedds-cxx/src/ddscxx/include/dds/topic/detail/TTopicImpl.hpp:312:77: error: ‘getTypeInfo’ is not a member of ‘org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::TParticipantBuiltinTopicData<org::eclipse::cyclonedds::topic::ParticipantBuiltinTopicDataDelegate> >’

This error appears 4 times.
Using the latest release of cyclonedds-cxx

**Comment by codingwei-lab** [comment]
Hi, brother. I have encountered a similar compilation problem. Have you solved this compilation error?

**Comment by NeilTruter** [comment]
> Hi, brother. I have encountered a similar compilation problem. Have you solved this compilation error?

Not yet

**Comment by codingwei-lab** [comment]
> > Hi, brother. I have encountered a similar compilation problem. Have you solved this compilation error?
> 
> Not yet

I have eliminated this error in other ways. You can try to compile the code using the release/0.9. x branch.

**Comment by reicheratwork** [comment]
Yeah, so this is caused by the `dds::topic::detail::Topic<T>::discover_topic` incorrectly assuming that the Topic Trait `getTypeInfo` is always available.
In the case that you compile CycloneDDS-CXX without Topic/Type discovery support, this function simply does not exist, that's why it's going wrong.
